File: library-base/network/src/main/java/com/ling/http/config/RequestServer.java
package com.ling.http.config;

import androidx.annotation.NonNull;

import com.ling.http.annotation.HttpIgnore;

/**
 * author : wangchengzhen
 * github : https://github.com/getActivity/EasyHttp
 * time   : 2022/5/19
 * desc   : 服务器简单配置
 */
public final class RequestServer implements IRequestServer {

    /** 主机地址 */
    @HttpIgnore
    private final String mHost;

    public RequestServer(String host) {
        mHost = host;
    }

    @NonNull
    @Override
    public String getHost() {
        return mHost;
    }

    @NonNull
    @Override
    public String toString() {
        return mHost;
    }
}

File: apps/app-mvc/src/main/java/com/ling/mvc/http/model/RequestServer.kt
package com.ling.mvc.http.model

import com.ling.http.config.IRequestServer
import com.ling.http.model.BodyType
import com.ling.mvc.config.AppConfig

/**
 * author : wangchengzhen
 * github : https://github.com/getActivity/AndroidProject-Kotlin
 * time   : 2022/6/8
 * desc   : 服务器配置
 */
class RequestServer : IRequestServer {

    override fun getHost(): String {
        return AppConfig.getHostUrl()
    }

    override fun getBodyType(): BodyType {
        // 以表单的形式提交参数
        return BodyType.FORM
    }
}
